TESTS/network/interface/networkinterface_tests.h

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2018, ARM Limited, All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef NETWORKINTERFACE_TESTS_H
+#define NETWORKINTERFACE_TESTS_H
+
+/*
+ * Test cases
+ */
+void NETWORKINTERFACE_STATUS();
```

Comment:
This is the naming convention for our socket and network tests. Test case names are in capital letters. Used in socket and dns tests already.

Reply:
Yes, older tests not using that yet. @SeppoTakalo should we update also those?

Reply:
I prefer the exact same name as test plan has.

* Test Plan https://github.com/ARMmbed/mbed-os/blob/0bb791ed0d1685b752712e652bd17e94895e3432/TESTS/netsocket/README.md
* https://github.com/ARMmbed/mbed-os/blob/master/TESTS/netsocket/tcp/tcp_tests.h
* https://github.com/ARMmbed/mbed-os/blob/master/TESTS/netsocket/udp/udp_tests.h
* https://github.com/ARMmbed/mbed-os/blob/master/TESTS/netsocket/dns/dns_tests.h

Test cases are anyway a bit special, so having exception to common API guidelines is acceptable.

Having testcases in small letters has a added benefit that it is clearly visible separation from test suite (binary name)
```
10:18:17 mbedgt: test suite 'mbed-os-tests-netsocket-dns' ..................................................... OK in 72.73 sec
10:18:17 	test case: 'ASYNCHRONOUS_DNS' ................................................................ OK in 0.11 sec
10:18:17 	test case: 'ASYNCHRONOUS_DNS_CACHE' .......................................................... OK in 0.33 sec
10:18:17 	test case: 'ASYNCHRONOUS_DNS_CANCEL' ......................................................... OK in 5.34 sec
10:18:17 	test case: 'ASYNCHRONOUS_DNS_EXTERNAL_EVENT_QUEUE' ........................................... OK in 2.60 sec
10:18:17 	test case: 'ASYNCHRONOUS_DNS_INVALID_HOST' ................................................... OK in 0.66 sec
10:18:17 	test case: 'ASYNCHRONOUS_DNS_NON_ASYNC_AND_ASYNC' ............................................ OK in 0.56 sec
10:18:17 	test case: 'ASYNCHRONOUS_DNS_SIMULTANEOUS' ................................................... OK in 0.55 sec
10:18:17 	test case: 'ASYNCHRONOUS_DNS_SIMULTANEOUS_CACHE' ............................................. OK in 0.55 sec
10:18:17 	test case: 'ASYNCHRONOUS_DNS_SIMULTANEOUS_REPEAT' ............................................ OK in 28.41 sec
10:18:17 	test case: 'ASYNCHRONOUS_DNS_TIMEOUTS' ....................................................... OK in 2.39 sec
```

Reply:
0xc0170 I think we would be changing the emac/wifi tests to match socket/dns tests in an another pull requests. Can this pull request proceed?

Reply:
Yes, entering needs: CI now.

> Test cases are anyway a bit special, so having exception to common API guidelines is acceptable.

It would be (I would still challenge we are entering the ground where preprocessor can replace it and leads to failures. I would avoid naming anything else with capital letters but macros) but this is not the case for test cases for our tests. I found this only in these tests not anywhere else so do not see why it would be different (we can update the test plan to match the implementation). If this is something up for the change, should be documented (test cases are upper case because..) and followed?

cc @ARMmbed/mbed-os-test